InterGalacticCurrencyConvertor/src/main/java/DataReader.java

```diff
@@ -8,56 +8,56 @@
  * Reads and processes lines of user input that can either be statements or
  * queries.
  */
-public class DataReader {
-	private BufferedReader mBufReader;
-	private DataProcessor mDataProcessor;
+public final class DataReader {
+	private final BufferedReader mBufReader;
+	private final DataProcessor mDataProcessor;
 	private static Boolean mDone = false;
 
-	public static void setExit(boolean inDone) {
-		mDone = inDone;
+	public static void setExit(boolean isDone) {
+		mDone = isDone;
 	}
-	public DataReader(InputStreamReader inStreamReader) {
-		mBufReader = new BufferedReader(inStreamReader);
-		mDataProcessor = new DataProcessor();
+	public DataReader(InputStreamReader streamReader) {
+		mBufReader = new BufferedReader(streamReader);
+		mDataProcessor = DataProcessor.getInstance();
 	}
-	private void printResult(DataProcessResult inResult) {
-	  Objects.requireNonNull(inResult, "Invalid Result Argument");
-		if (inResult.getOutput() != null) {
-			System.out.println(inResult.getOutput());
+	private void printResult(DataProcessResult result) {
+	  Objects.requireNonNull(result, "Invalid Result Argument");
+		if (result.getProcessorResult() != null) {
+			System.out.println(result.getProcessorResult());
 		}
 	}
 	public String readData() {
-		String vLine = null;
+		String line = null;
 		try {
-			vLine = mBufReader.readLine();
+			line = mBufReader.readLine();
 		} catch (IOException e) {
 			mDone = true;
 		}
-		return vLine;
+		return line;
 	}
-	public DataProcessResult processData(String inLine) {
-		return mDataProcessor.processData(inLine);
+	public DataProcessResult processData(String line) {
+		return mDataProcessor.processData(line);
 	}
 	public static void main(String [] args) {
-		DataReader vReader = null;
-		DataProcessResult vResult = null;
-		InputStreamReader vStreamReader = new InputStreamReader(System.in);
+		DataReader reader = null;
+		DataProcessResult result = null;
+		InputStreamReader streamReader = new InputStreamReader(System.in);
 		try {
```

Comment:
Can you simplify the line reading code using BufferedReader - maybe it can be all included in the main(...) method? Also, using try { ...} around the BufferedReader so that you don't have to do cleaning manually later. Look at the following:

https://stackoverflow.com/questions/5868369/how-to-read-a-large-text-file-line-by-line-using-java

Reply:
this was done so that the input is not limited to system.in and for testing purpose